tests/components/qbus/test_config_flow.py

```diff
@@ -0,0 +1,249 @@
+    discovery = MqttServiceInfo(
+        subscribed_topic="cloudapp/QBUSMQTTGW/state",
+        topic="cloudapp/QBUSMQTTGW/state",
+        payload='{ "online": false }',
+        qos=0,
+        retain=False,
+        timestamp=time.time(),
+    )
+
+    with (
+        patch("homeassistant.components.mqtt.client.async_publish") as mock_publish,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_MQTT}, data=discovery
+        )
+
+    assert mock_publish.called is False
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "discovery_in_progress"
+
+
+async def test_handle_config_topic(hass: HomeAssistant) -> None:
+    """Test handling of config topic."""
+    discovery = MqttServiceInfo(
+        subscribed_topic=TOPIC_CONFIG,
+        topic=TOPIC_CONFIG,
+        payload=PAYLOAD_CONFIG,
+        qos=0,
+        retain=False,
+        timestamp=time.time(),
+    )
+
+    with (
+        patch.object(
+            QbusConfigCoordinator, "store_config", return_value=None, autospec=True
+        ) as mock_store,
+        patch("homeassistant.components.mqtt.client.async_publish") as mock_publish,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_MQTT}, data=discovery
+        )
+
+    assert mock_store.called
+    assert mock_publish.called
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "discovery_in_progress"
+
+
+async def test_handle_device_topic_missing_config(hass: HomeAssistant) -> None:
+    """Test handling of device topic when config is missing."""
+    discovery = MqttServiceInfo(
+        subscribed_topic="cloudapp/QBUSMQTTGW/+/state",
+        topic="cloudapp/QBUSMQTTGW/UL1/state",
+        payload="{ }",
+        qos=0,
+        retain=False,
+        timestamp=time.time(),
+    )
+
+    with (
+        patch.object(
+            QbusConfigCoordinator,
+            "async_get_or_request_config",
+            return_value=None,
+        ) as mock_get_config,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_MQTT}, data=discovery
+        )
+
+    assert mock_get_config.called
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "invalid_discovery_info"
+
+
+async def test_handle_device_topic_config_not_ready(hass: HomeAssistant) -> None:
+    """Test handling of device topic when device is not found."""
+    discovery = MqttServiceInfo(
+        subscribed_topic="cloudapp/QBUSMQTTGW/+/state",
+        topic="cloudapp/QBUSMQTTGW/UL1/state",
+        payload='{"id":"UL1","properties":{"connected":true},"type":"event"}',
+        qos=0,
+        retain=False,
+        timestamp=time.time(),
+    )
+
+    with patch.object(
+        QbusConfigCoordinator,
+        "async_get_or_request_config",
+        return_value=None,
```

Comment:
isn't this test failing for the same reason as the one above (`async_get_or_request_config` returns `None`)?

Reply:
Oops, this is indeed a duplicate test. I removed it.